internal/repository/master_index.go

```diff
@@ -22,18 +22,30 @@ func NewMasterIndex() *MasterIndex {
 }
 
 // Lookup queries all known Indexes for the ID and returns the first match.
-func (mi *MasterIndex) Lookup(id restic.ID, tpe restic.BlobType) (blobs []restic.PackedBlob, found bool) {
+func (mi *MasterIndex) Lookup(id restic.ID, tpe restic.BlobType) (blob restic.PackedBlob, found bool) {
 	mi.idxMutex.RLock()
 	defer mi.idxMutex.RUnlock()
 
 	for _, idx := range mi.idx {
-		blobs, found = idx.Lookup(id, tpe)
+		blob, found = idx.Lookup(id, tpe)
 		if found {
 			return
 		}
 	}
 
-	return nil, false
+	return restic.PackedBlob{}, false
+}
+
+// Lookup queries all known Indexes for the ID and returns all matches.
+func (mi *MasterIndex) LookupAll(id restic.ID, tpe restic.BlobType) (blobs []restic.PackedBlob) {
+	mi.idxMutex.RLock()
+	defer mi.idxMutex.RUnlock()
+
+	for _, idx := range mi.idx {
+		blobs = append(blobs, idx.LookupAll(id, tpe)...)
```

Comment:
There's a potential for optimization here: if Index.LookupAll appends instead of always returning a new list, then this loop can be done with fewer allocations. Viz.,

```go
 func (mi *MasterIndex) LookupAll(id restic.ID, tpe restic.BlobType, blobs []restic.PackedBlob) []restic.PackedBlob {
	// lock+unlock
	for _, idx := range mi.idx {
		blobs = idx.LookupAll(id, tpe, blobs)
	}
	return blobs
}
```

Reply:
See aawsome/restic#2.

Reply:
Thank you for pointing this out! However, I'm not confident to also changing `MasterIndex.LookupAll` as this would be usually not called multiple times.

I just changed `Index.LookupAll`.